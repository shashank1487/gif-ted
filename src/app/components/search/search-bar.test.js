import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import SearchBar from "./search-bar";

configure({ adapter: new Adapter() });
describe("The search bar component", () => {
  let app;
  let mockSearchHandler = jest.fn();
  let mockChangeHandler = jest.fn();
  
  beforeEach(() => {
    app = shallow(
      <SearchBar
        searchTerm="obama"
        changeHandler={mockChangeHandler}
        searchHandler={mockSearchHandler}
      />
    );
  });

  it('should render correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it("the search button should have text", () => {
    let button = app.find("button").first();
    expect(button.text()).toEqual("Search");
  });

  it("should handle the click event", () => {
    app.find("button").simulate("click");
    expect(mockSearchHandler.mock.calls.length).toEqual(1);
  });

  it("should handle the change event", () => {
    app.find("input[type='text']").simulate("change");
    expect(mockChangeHandler.mock.calls.length).toEqual(1);
  });
});
