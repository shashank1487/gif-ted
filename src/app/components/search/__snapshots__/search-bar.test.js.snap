// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The search bar component should render correctly 1`] = `ShallowWrapper {}`;
